packages/components/src/popover/index.js

```diff
@@ -259,6 +346,10 @@ const Popover = ( {
 	position = 'top',
 	range,
 	focusOnMount = 'firstElement',
+	anchorRef,
+	anchorWithoutPadding,
+	anchorVerticalBuffer,
+	anchorHorizontalBuffer,
```

Comment:
Do you think we should have a single prop instead with different options? `anchor = { ref, padding, buffer }`

Reply:
I don't like nesting that much for component props. That would break shallow comparison, right? Maybe we can name them differently...

Reply:
It does break shallow comparison. we can use things like `useMemo` to avoid that but I'm not sure if it's worth it.

I like the approach of this PR, maybe we can get more thoughts on props naming here @mcsf

Reply:
I wonder how other popover libraries or design systems name/handle the positioning options. cc @ItsJonQ

Reply:
@youknowriad + @ellatrix 👋 Halloo!

A popular popover/positioner library folks use (under the hood) is Popper.js

Their React implementation uses the main library and share the same API.
They use nested objects, example:

`offsets.popper`
top, left, width, height values

`data.offsets.reference`
top, left, width, height values

I've used both shallow and nested approaches before. Both have their pros and cons :).

Typically component library components/primitives have *lots* of props. Way more compared to components created at the app level.

If this were going into`@wordpress/components` to update the `Popover`, I would vote for @youknowriad 's suggestion with **nested**.

However, since this is updating an "application" component, something that typically won't be consumed externally and used to build interfaces... I would agree with you @ellatrix, and vote for **shallow**.

Hope this helps! <3

Reply:
> I've used both shallow and nested approaches before. Both have their pros and cons :).

Could you elaborate? What are the pros of nesting? I'm glad to see that `Popper` also requires an element reference, just as I'm proposing here.

Reply:
@ellatrix 

> I'm glad to see that Popper also requires an element reference, just as I'm proposing here.

❤️ 

A pro of nesting would be that it keeps related props grouped together.
Also makes them easier to work with (IMO) those props are passed within a [render prop](https://reactjs.org/docs/render-props.html).

I would only nest 1 level deep (2 max!!! if you really really need to)

An example would be [react-select](https://react-select.com/props). The `components` prop is a good one. It allows the user to pass in custom components to replace their sub-components (e.g. `Menu`). If that wasn't there, they might need to define something like `componentMenu`, `componentInput`, etc...

It's not the end of the world. But that library has *a ton* of props. I think this keeps it tidier :)

That being said.. if one decides to nest, they'll need to do things like (as @youknowriad mentioned) memoize. Also to do some sort of defaultValues + userDefinedValues merging.

Reply:
`useMemo` puts extra responsibilities on the implementor. Not sure if it's worth it for these few props.

Reply:
🤦‍♂ My apologies! I didn't realize these changes were on `@wordpress/components/popover`.
(It was early for me)

> useMemo puts extra responsibilities on the implementor. 

I agree. These primitives should take care of this overhead.

If we were to use an `object`, would exporting it like this help?

```
import { useRef, useState, useEffect, memo } from '@wordpress/element';
...
const PopoverContainer = memo(Popover);
```

I'm guessing not, since `memo` only does shallow comparisons.

>  Not sure if it's worth it for these few props.

Given its current state, I think it might be better to keep it simple (and shallow). 
My concern was maybe later the number of props + names of them may get unwieldy, but that may not happen :).

Apologies for the back and forth!

Reply:
I mean, the interface provider can always manage itself with `shouldComponentUpdate`, and I get the benefit of keeping things tidy with nested props.

The question is probably how much we expect these interfaces to grow, since we aren't building an all-encompassing popover library. My first impression is that it's probably not worth the extra complexity to have nested props.

> I like the approach of this PR, maybe we can get more thoughts on props naming here @mcsf

I think this PR is good. I would just double-check that the props are named in a way that conveys their type well, as pointed out in https://github.com/WordPress/gutenberg/pull/17867/files#r351238110.